### DISCUSSION TITLE
magit and bookmarks: autoload needed?

### DISCUSSION THREAD

**Discussion by jdtsmith** [magit/magit#5168] [Ideas]
I'm using [activities](https://github.com/alphapapa/activities.el) by @alphapapa, which uses bookmarks to restore buffers and windows.  I tend to accumulate magit buffers among my activity's windows and have always wished magit could restore them.  To my surprise I discovered `magit--handle-bookmark` already exists!  

Magit buffer/window recovery does not currently function with activities after a start, however, since `magit--handle-bookmark` isn't loaded.  I think an `autoload` to load `magit-section` would fix this.  Probably the `cl-defmethod`'s in `magit-bookmark` also need to be loaded.

**Comment by tarsius** [comment]
That seems worth supporting.

Give this a try:

```diff
diff --git a/lisp/magit-section.el b/lisp/magit-section.el
index 3432fa4a32..0e27357e78 100644
--- a/lisp/magit-section.el
+++ b/lisp/magit-section.el
@@ -2396,6 +2396,7 @@ (defun magit--make-bookmark ()
         bookmark)
     (user-error "Bookmarking is not implemented for %s buffers" major-mode)))
 
+;;;###autoload
 (defun magit--handle-bookmark (bookmark)
   "Open a bookmark created by `magit--make-bookmark'.
 
@@ -2405,6 +2406,7 @@ (defun magit--handle-bookmark (bookmark)
 Then call the `magit-*-setup-buffer' function of the the major-mode
 with the variables' values as arguments, which were recorded by
 `magit--make-bookmark'."
+  (require 'magit-bookmark nil t)
   (let ((buffer (magit-bookmark-get-buffer-create
                  bookmark
                  (bookmark-prop-get bookmark 'mode))))
```

and if it works, open a pull-request, with a message that explains why we are doing this.

**Comment by jdtsmith** [comment]
Thanks, that works perfectly  (once I ran `(package-generate-autoloads "magit-section" "~/.emacs.d/elpa/magit-section-20240613.2254")`).

**Comment by jdtsmith** [comment]
See PR #5169.
